### DISCUSSION TITLE
Electron App - Use SQLite or CSV Files

### DISCUSSION THREAD

**Discussion by byander** [quasarframework/quasar#14147] [CLI - Electron mode] [answered]
Hello,
I'm creating a offline desktop application with Quasar v2 and Electron, following the guide from oficial documentation.

In the application I need use some data, so, I've been trying without success use data from sqlite database or csv files to show on tables in the application.
I tried use de packages like sqlite3, better-sqlite and csvtojson, but always I get error.
I think that the package csvtojson already solves my problem, but I get the error: Cannot find module 'fs'.
How I fix this?
I not find a guide or tutorial for use the framework Quasar with data from SQLite or csv files.

Thanks.

**Comment by TobyMosque** [comment, marked as answer]
To use NodeJS related packages, u would be in the main process, not in the renderer process.

You had 3 options:

1 - Preload Script - put all your code who requires NodeJS stuff in the preload script, so call them from the renderer process.

https://quasar.dev/quasar-cli-vite/developing-electron-apps/electron-preload-script#introduction

2 - Main Process - put all your code who requires NodeNS in the main process, so use IPC (Main & Renderer) to allow the Renderer Process call the main Process.

https://www.electronjs.org/docs/latest/tutorial/ipc

3 - Browser Friendly DB - use a db library who is browser friendly, like PouchDB, LokiJS, Dexie, etc.

https://dexie.org/
https://github.com/techfort/LokiJS
https://pouchdb.com/

Personally, i would pick the 3th option with the most abstract one database lib, namely, RxDB

https://rxdb.info/rx-storage-dexie.html

**Reply by byander** [reply]
Thanks for your help.
I decided use the first option (preload) and it's works now.
